crates/hash/src/groestl/arch/groestl_multi_avx2.rs

```diff
@@ -0,0 +1,431 @@
+// Copyright 2025 Irreducible Inc.
+
+use std::arch::x86_64::*;
+pub type State = [__m256i; 8];
```

Comment:
Maybe add a new structure using [new type pattern](https://doc.rust-lang.org/rust-by-example/generics/new_types.html), so you convert free functions to the methods of this structure?